modules/custom/social_eda/src/Types/ContentVisibility.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\social_eda\Types;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\social_role_visibility\Service\VisibilityElementManager;
+
+/**
+ * Type class for ContentVisibility data.
+ */
+class ContentVisibility {
+
+  /**
+   * Constructs the ContentVisibility type.
+   */
+  public function __construct(
+    public readonly string $type,
+    public readonly array $groups = [],
+    public readonly array $roles = [],
+  ) {}
+
+  /**
+   * Get formatted ContentVisibility output.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The entity object.
+   *
+   * @return self|null
+   *   The ContentVisibility object or null.
+   */
+  public static function fromEntity(ContentEntityInterface $entity): ?self {
+    if (!$entity->hasField('field_content_visibility')) {
+      return NULL;
+    }
+
+    $visibility_type = $entity->get('field_content_visibility')->value;
+
+    switch ($visibility_type) {
+      case "group":
+        $groups_ids = [];
+        if ($entity->hasField('groups')) {
+          /** @var \Drupal\group\Entity\GroupInterface $groups */
+          $groups = $entity->get('groups');
+          foreach ($groups->referencedEntities() as $group) {
+            $groups_ids[] = $group->uuid();
+          }
+        }
+
+        return new self(
+          type: (string) $visibility_type,
```

Comment:
@nkoporec, as I can see in the Payload design, type should be `groups` and not `group`. 

Payload:
![image](https://github.com/user-attachments/assets/e3ef36c2-0629-4de0-bf04-6e855d0d6c1a)

Current:
<img width="406" alt="image" src="https://github.com/user-attachments/assets/5d7b7e2c-83db-4ca0-90e6-bc162e0023e7">